src/dragonball/src/hypervisor_metrics.rs

```diff
@@ -75,28 +78,33 @@ fn update_hypervisor_metrics() -> Result<()> {
 }
 
 fn set_intgauge_vec_vcpu(icv: &prometheus::IntGaugeVec) {
-    icv.with_label_values(&["exit_io_in"])
-        .set(METRICS.vcpu.exit_io_in.count() as i64);
-    icv.with_label_values(&["exit_io_out"])
-        .set(METRICS.vcpu.exit_io_out.count() as i64);
-    icv.with_label_values(&["exit_mmio_read"])
-        .set(METRICS.vcpu.exit_mmio_read.count() as i64);
-    icv.with_label_values(&["exit_mmio_write"])
-        .set(METRICS.vcpu.exit_mmio_write.count() as i64);
-    icv.with_label_values(&["failures"])
-        .set(METRICS.vcpu.failures.count() as i64);
-    icv.with_label_values(&["filter_cpuid"])
-        .set(METRICS.vcpu.filter_cpuid.count() as i64);
+    let metric_gurad = METRICS.read().unwrap();
+    for (cpu_id, metrics) in metric_gurad.vcpu.iter() {
+        icv.with_label_values(&[cpu_id.to_string().as_str(), "exit_io_in"])
+            .set(metrics.exit_io_in.count() as i64);
+        icv.with_label_values(&[cpu_id.to_string().as_str(), "exit_io_out"])
+            .set(metrics.exit_io_out.count() as i64);
+        icv.with_label_values(&[cpu_id.to_string().as_str(), "exit_mmio_read"])
+            .set(metrics.exit_mmio_read.count() as i64);
+        icv.with_label_values(&[cpu_id.to_string().as_str(), "exit_mmio_write"])
+            .set(metrics.exit_mmio_write.count() as i64);
+        icv.with_label_values(&[cpu_id.to_string().as_str(), "failures"])
+            .set(metrics.failures.count() as i64);
+        icv.with_label_values(&[cpu_id.to_string().as_str(), "filter_cpuid"])
+            .set(metrics.filter_cpuid.count() as i64);
+    }
 }
 
 fn set_intgauge_vec_seccomp(icv: &prometheus::IntGaugeVec) {
+    let metric_gurad = METRICS.read().unwrap();
```

Comment:
typo? metric_guard?

Reply:
It's a  `RwLockReadGuard` for RwLock<DragonballMetrics>.

Reply:
I know,but it should be `metric_guard` instead of `metric_gurad`

Reply:
Thanks. Maybe I should change my eyes : )